metrics/codahale.go

```diff
@@ -178,14 +176,34 @@ func (c *CodaHale) MeasureResponse(code int, method string, routeId string, star
 }
 
 func (c *CodaHale) MeasureServe(routeId, host, method string, code int, start time.Time) {
+	if !(c.options.EnableServeRouteMetrics || c.options.EnableServeHostMetrics) {
+		return
+	}
+
+	var keyServeRoute, keyServeHost string
 	method = measuredMethod(method)
+	hfk := hostForKey(host)
+	switch {
```

Comment:
Here it may be similar to prometheus using nil slice trick and strings.Join:
```diff
-       var keyServeRoute, keyServeHost string
-       method = measuredMethod(method)
-       hfk := hostForKey(host)
-       switch {
-       case c.options.EnableServeMethodMetric && c.options.EnableServeStatusCodeMetric:
-               keyServeHost = fmt.Sprintf("servehost.%s.%s.%d", hfk, method, code)
-               keyServeRoute = fmt.Sprintf("serveroute.%s.%s.%d", routeId, method, code)
-       case c.options.EnableServeMethodMetric:
-               keyServeHost = fmt.Sprintf("servehost.%s.%s", hfk, method)
-               keyServeRoute = fmt.Sprintf("serveroute.%s.%s", routeId, method)
-       case c.options.EnableServeStatusCodeMetric:
-               keyServeHost = fmt.Sprintf("servehost.%s.%d", hfk, code)
-               keyServeRoute = fmt.Sprintf("serveroute.%s.%d", routeId, code)
-       default:
-               keyServeHost = fmt.Sprintf("servehost.%s", hfk)
-               keyServeRoute = fmt.Sprintf("serveroute.%s", routeId)
+       var metrics []string
+       if c.options.EnableServeStatusCodeMetric {
+               metrics = append(metrics, fmt.Sprint(code))
+       }
+       if c.options.EnableServeMethodMetric {
+               metrics = append(metrics, measuredMethod(method))
        }
 
        if c.options.EnableServeRouteMetrics {
-               c.measureSince(keyServeRoute, start)
+               c.measureSince(strings.Join(append([]string{"serveroute", routeId}, metrics...), "."), start)
        }
 
        if c.options.EnableServeHostMetrics {
-               c.measureSince(keyServeHost, start)
+               c.measureSince(strings.Join(append([]string{"servehost", hostForKey(host)}, metrics...), "."), start)
        }
```

Reply:
I think this way will be very similar to the way it was before. I'm fine with both solutions at this moment and I don't think any of them will have a huge increase in performance. I will let it for you and @aryszka decide.